go/runtime/bundle/manager.go

```diff
@@ -437,10 +473,61 @@ func (m *Manager) fetchBundle(url string) (string, error) {
 	return file.Name(), nil
 }
 
-func (m *Manager) loadManifests() (map[string]*Manifest, error) {
+func (m *Manager) cleanBundles(runtimeID common.Namespace) {
+	m.mu.Lock()
+	maxVersion, ok := m.cleanupQueue[runtimeID]
+	if !ok {
+		m.mu.Unlock()
+		return
+	}
+	// Do not retry clean-up if it fails.
+	delete(m.cleanupQueue, runtimeID)
+	m.mu.Unlock()
+
+	m.logger.Info("starting clean-up of bundles",
+		"id", runtimeID,
+		"max_version", maxVersion,
+	)
+
+	for _, manifest := range m.store.Manifests() {
+		if manifest.ID != runtimeID {
+			continue
+		}
+
+		if manifest.IsDetached() {
+			continue
+		}
+
+		if ronl := manifest.GetComponentByID(component.ID_RONL); !ronl.Version.Less(maxVersion) {
+			continue
+		}
+
+		m.cleanStaleBundle(manifest)
+	}
+}
+
+func (m *Manager) cleanStaleBundle(manifest *ExplodedManifest) {
+	m.logger.Info("cleaning bundle",
+		"manifest_hash", manifest.Hash().Hex(),
+	)
+
+	if ok := m.store.RemoveManifest(manifest.Hash()); !ok {
+		m.logger.Debug("manifest already removed from the store",
```

Comment:
```suggestion
		m.logger.Debug("failed to remove manifest from store",
```
Message is not true in general. Maybe the manifest was never in the store.